### PR TITLE
parser: add testcases for lexer

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

**Comment by shenli** [comment]
LGTM
@coocood PTAL

**Comment by coocood** [comment]
LGTM

File: parser/scanner_test.go [hunk 1]
```diff
@@ -0,0 +1,444 @@
+// Copyright 2015 PingCAP, Inc.
```
Comment (zimulala) [inline parser/scanner_test.go]: 2015 -> 2016

**Comment by zimulala** [comment]
LGTM

**Pull request merged**